query-engine/query-engine-node-api/src/node_drivers/engine.rs

```diff
@@ -0,0 +1,140 @@
+
+pub struct NodejsFunctionContext {
+    env: Env,
+    pub query_raw: JsFunction,
+    pub query_raw_typed: JsFunction,
+    pub execute_raw: JsFunction,
+    pub execute_raw_typed: JsFunction,
+    pub version: JsFunction,
+}
+
+impl NodejsFunctionContext {
+    pub fn query_raw(&self) -> napi::Result<()> {
+        let arg = self.env.create_string("SELECT * FROM users").unwrap();
+        let _promise: JsUnknown = self.query_raw.call(None, &[arg])?;
+        // TODO: cast JsUnknown to JsPromise
+
+        Ok(())
+    }
+
+    pub fn query_raw_typed(&self) -> napi::Result<()> {
+        let arg = self.env.create_string("SELECT * FROM users").unwrap();
+        let _promise: JsUnknown = self.query_raw_typed.call(None, &[arg])?;
+        // TODO: cast JsUnknown to JsPromise, idk how yet :/
+
+        Ok(())
+    }
+
+    pub fn execute_raw(&self) -> napi::Result<()> {
+        let arg = self.env.create_string("EXECUTE ...").unwrap();
+        let _promise: JsUnknown = self.execute_raw.call(None, &[arg])?;
+        // TODO: cast JsUnknown to JsPromise, idk how yet :/
+
+        Ok(())
+    }
+
+    pub fn execute_raw_typed(&self) -> napi::Result<()> {
+        let arg = self.env.create_string("EXECUTE ...").unwrap();
+        let _promise: JsUnknown = self.execute_raw_typed.call(None, &[arg])?;
+        // TODO: cast JsUnknown to JsPromise, idk how yet :/
+
+        Ok(())
+    }
+
+    pub fn version(&self) -> napi::Result<String> {
+        let version_raw_str: JsUnknown = self.version.call::<JsUnknown>(None, &[])?;
+        let version_js_str: JsString = version_raw_str.try_into()?;
+        let version_str = version_js_str.into_utf8()?.into_owned()?;
+
+        Ok(version_str)
+    }
+}
+
+fn read_nodejs_functions<'a>(env: Env, ctx: JsObject) -> napi::Result<NodejsFunctionContext> {
+    let query_raw = ctx.get_named_property("queryRaw")?;
+    let query_raw_typed = ctx.get_named_property("queryRawTyped")?;
+    let execute_raw = ctx.get_named_property("executeRaw")?;
+    let execute_raw_typed = ctx.get_named_property("executeRawTyped")?;
+    let version = ctx.get_named_property("version")?;
+
+    let ctx = NodejsFunctionContext {
+        env,
+        query_raw,
+        query_raw_typed,
+        execute_raw,
+        execute_raw_typed,
+        version,
+    };
+    Ok(ctx)
+}
+
+// Wrapper for the main query engine that keeps track of the JS-provided functions
+// used to execute queries.
+#[napi]
+pub struct QueryEngineNodeDrivers {
+    pub(crate) engine: QueryEngine,
+
+    #[allow(dead_code)]
+    pub(crate) fn_ctx: NodejsFunctionContext,
+}
+
+#[napi]
+impl QueryEngineNodeDrivers {
+    /// Parse a validated datamodel and configuration to allow connecting later on.
+    /// Wraps the `QueryEngine::new` constructor.
+    #[napi(constructor)]
+    pub fn new(napi_env: Env, options: JsUnknown, callback: JsFunction, fn_ctx: JsObject) -> napi::Result<Self> {
+        let fn_ctx = read_nodejs_functions(napi_env.clone(), fn_ctx)?;
```

Comment:
This `fn_ctx` can be used to instantiate the global `Queryable` implementation we were discussing about